docs/modules/ROOT/pages/contributors/naming-conventions.adoc

```diff
@@ -0,0 +1,126 @@
+= Naming Conventions
+
+This document lists the naming conventions to be applied in the different parts
+of KubeArchive codebase.
+
+== Helm Chart
+
+The Helm Chart templates are located into different directories that match the different
+components of kubearchive:
+
+* Sink
+* API Server
+* Operator
+* Database
+
+The common resources are outside these folders.
+
+The templated variables are located in [filename]`kubearchive/chars/kubearchive/values.yaml`
+following the same categories.
+
+=== Conventions
+
+* Follow the DRY (Don't Repeat Yourself) principle.
+When a value is reused extract it and reference it.
+If a part of the value is reused (like a prefix with the app name) also
+reuse it.
++
+There are 4 possibilities for referencing variables and not repeat them:
+
+** Direct reference to a variable in [filename]`values.yaml`.
++
+.values.yaml
+[source,yaml]
+----
+apiServer:
+  port: 8081
+----
++
+.template-example.yaml
+[source,yaml]
+----
+kind: Service
+apiVersion: v1
+spec:
+  ports:
+    - protocol: TCP
+      port: {{ .Values.apiServer.port }}
+      targetPort: {{ .Values.apiServer.port }}
+      name: server
+----
```

Comment:
I think that adding something to the file values.yaml shouldn't be listed as a DRY mechanism, ideally the values.yaml should contains just things for the user.

Reply:
You are right, and although the main point of this conventions is to use variables to reuse things, it's easy to expose this idea, so I've removed the DRY part and just stated examples of what we are doing and the tools that we are using. Please take a look to the changes.